### PR TITLE
filter by category

### PR DESCRIPTION
**Pull request by khrystyna-pylypchuk**

### CODE REVIEW

**Review by skrechko — APPROVED** [review]

File: src/app/search-page/search-page.component.ts [hunk 1]
```diff
@@ -42,12 +57,23 @@ export class SearchPageComponent implements OnInit, OnDestroy {
...
   }
 
-  onGetCocktails (data) {
-    if (data.length) {
-      this.filteredByIngredientCocktails = data;
-    } else {
-      this.filteredByIngredientCocktails = this.cocktailsList;
+  onGetCocktails ({data, category}) {
+    const isEl = this.fullCocktailsLists.find(el => {
+      if (el.category === category) {
```
Comment (dosandk) [inline src/app/search-page/search-page.component.ts]: can we move this expression to some variable? ```const isEqualCategories = el.category === category;```

File: src/services/cocktails.service.ts [hunk 2]
```diff
@@ -3,18 +3,17 @@ import { HttpClient  } from '@angular/common/http';
...
-
 @Injectable()
 export class CocktailsService {
   constructor (private http: HttpClient) {}
 
-  getCocktails(): Observable<CocktailsList> {
-    const response = this.http.get(DATA_URL.COCKTAILS);
-    return response.pipe(pluck(DATA_KEY));
+  getCocktails(category): Observable<CocktailsList> {
+    const response = this.http.get(DATA_URL.COCKTAILS + category);
```
Comment (dosandk) [inline src/services/cocktails.service.ts]: will be great to replace this by string interpolation ```const response = this.http.get(`DATA_URL.COCKTAILS category`);```

**Review by dosandk — APPROVED** [review]

File: src/services/cocktails.service.ts [hunk 3]
```diff
@@ -31,4 +30,9 @@ export class CocktailsService {
     const response = this.http.get(`${DATA_URL.COCKTAILS_BY_INGREDIENT}${ingredient}`);
     return response.pipe(pluck(DATA_KEY));
   }
+
+  getCategories(): Observable<Categories> {
+    const response = this.http.get('//www.thecocktaildb.com/api/json/v1/1/list.php?c=list');
```
Comment (ntsyhaniuk) [inline src/services/cocktails.service.ts]: You can move this url also into config

**Review by ntsyhaniuk — APPROVED** [review]

File: src/services/cocktails.service.ts (hunk 2 above)
Comment (khrystyna-pylypchuk) [reply src/services/cocktails.service.ts]: done

File: src/app/search-page/search-page.component.ts (hunk 1 above)
Comment (khrystyna-pylypchuk) [reply src/app/search-page/search-page.component.ts]: done

File: src/services/cocktails.service.ts (hunk 3 above)
Comment (khrystyna-pylypchuk) [reply src/services/cocktails.service.ts]: done

**Pull request merged**